cmd/mdatagen/internal/samplereceiver/metadata.yaml

```diff
@@ -174,3 +174,12 @@ telemetry:
         async: true
         value_type: int
         monotonic: true
+    optional_metric:
```

Comment:
Can you use a different name here? It wasn't very apparent to me first whether "optional_metric" is a reserved word for this feature or the metric name. It became clear when I saw "optional: true" below, but perhaps you can use a real metric name instead?

Reply:
Thanks for the suggestion, i changed it to queue_length, which is sorta what this PR is all about 😆